### DISCUSSION TITLE
Why CoreWCF defines its own XmlSerializerFormatAttribute,

### DISCUSSION THREAD

**Discussion by JohnLeyva** [CoreWCF/CoreWCF#372] [General]
CoreWCF defines its own CoreWCF.XmlSerializerFormatAttribute.

Wondering, why it does not use the  [System.ServiceModel.XmlSerializerFormatAttribute], as it seems to provide the same properties.

This duplcation is forcing the use of the following befere interfaces and methods.

[CoreWCF.XmlSerializerFormat]
[System.ServiceModel.XmlSerializerFormat]

Also, is good to consider that svcutil generates System.ServiceModel.XmlSerializerFormat

**Comment by mconnew** [comment]
The reason is to avoid pulling in the entire WCF client nuget package unnecessarily. We currently have compatibility code using reflection to convert some attributes in the System.ServiceModel namespace to the equivalent ones in the CoreWCF namespace. This looks like it's a good candidate for the same approach. I'll create an issue to track this.

**Reply by mconnew** [reply]
@JohnLeyva, I've open issue #378 to track adding this support. If this is something you would like to work on, let me know in the issue and mention me and I'll point you to the existing compatibility code for other attributes and give you some guidance on the best place to add code to support the ServiceModel version of the attribute.

**Reply by g7ed6e** [reply]
This is now implemented. @mconnew i was thinking about the compat code and I thought we may use source generators to enrich user provided contracts. This way compat code could be removed.

**Reply by JohnLeyva** [reply]
Code Generators will indeed facilitate the transition to CoreWCF

**Comment by JohnLeyva** [comment]
With 0.2.1, is still required to do the following to fix svcutils generated files

1. Include the following at the top

#if NETFRAMEWORK
using System.ServiceModel;
#else
using CoreWCF;
#endif

2. Remove all the "System.ServiceModel." in the file.

3. Remove Generated Client

**Reply by g7ed6e** [reply]
No, this is no longer required. This unit test https://github.com/CoreWCF/CoreWCF/blob/f6c9f74793fc7c94eb7be8a20886e73d63d3c45d/src/CoreWCF.Http/tests/ServiceWithSSMXmlSerializerFormatTest.cs covers that case.

**Reply by g7ed6e** [reply]
@JohnLeyva my bad this is part of the 0.3.0 not the 0.2.1.
